ext/curl/curl_url.stub.php

```diff
@@ -0,0 +1,240 @@
+<?php
+
+/** @generate-class-entries */
+
+#if LIBCURL_VERSION_NUM >= 0x073E00 /* Available since 7.62.0 */
+/**
+ * @strict-properties
+ * @not-serializable
+ */
+final class CurlUrl implements Stringable
+{
+    /**
+     * @var int
+     * @cname CURLU_APPENDQUERY
+     */
+    public const APPENDQUERY = UNKNOWN;
+
+    /**
+     * @var int
+     * @cname CURLU_DEFAULT_PORT
+     */
+    public const DEFAULT_PORT = UNKNOWN;
+
+    /**
+     * @var int
+     * @cname CURLU_DEFAULT_SCHEME
+     */
+    public const DEFAULT_SCHEME = UNKNOWN;
+
+    /**
+     * @var int
+     * @cname CURLU_DISALLOW_USER
+     */
+    public const DISALLOW_USER = UNKNOWN;
+
+    /**
+     * @var int 
+     * @cname CURLU_GUESS_SCHEME
+     */
+    public const GUESS_SCHEME = UNKNOWN;
+
+    /**
+     * @var int
+     * @cname CURLU_NO_DEFAULT_PORT
+     */
+    public const NO_DEFAULT_PORT = UNKNOWN;
+
+    /**
+     * @var int
+     * @cname CURLU_NON_SUPPORT_SCHEME
+     */
+    public const NON_SUPPORT_SCHEME = UNKNOWN;
+
+    /**
+     * @var int
+     * @cname CURLU_PATH_AS_IS
+     */
+    public const PATH_AS_IS = UNKNOWN;
+
+    /**
+     * @var int
+     * @cname CURLU_URLDECODE
+     */
+    public const URLDECODE = UNKNOWN;
+
+    /**
+     * @var int
+     * @cname CURLU_URLENCODE
+     */
+    public const URLENCODE = UNKNOWN;
+
+#if LIBCURL_VERSION_NUM >= 0x074300 /* Available since 7.67.0 */
+    /**
+     * @var int
+     * @cname CURLU_NO_AUTHORITY
+     */
+    public const NO_AUTHORITY = UNKNOWN;
+#endif
+      
+#if LIBCURL_VERSION_NUM >= 0x074e00 /* Available since 7.78.0 */
+    /**
+     * @var int
+     * @cname CURLU_ALLOW_SPACE
+     */
+    public const ALLOW_SPACE = UNKNOWN;
+#endif
+
+    public function __construct(?string $url = null, int $flags = 0) {}
+
+    public function get(int $flags = 0): string {}
+    public function set(?string $url, int $flags = 0): CurlUrl {}
```

Comment:
It can be used to change all the values, but it also works with relatives url. You can see an example of its usage in `curl_url_020.phpt`.

Reply:
Hm, OK; so not symmetric with `get` then? Perhaps something along the lines of "setMultipleParts" would describe it better? Or even just "setRelative", since in practice passing an absolute URL to it wouldn't make much sense?

Reply:
I think `setMultipleParts` is better since it not supports only relative urls. It does all the same as `__construct` but also supports relative urls.